### PR TITLE
[NO-2405] Allow internal setting of a `_priceMultiple` state variable, getter

### PR DESCRIPTION
**Pull request by amiecorso**
Establishes a state variable in the `Market` contract, `_priceMultiple`, which can later be used as a multiple of the underlying base token to determine how many of that token are required to purchase one NRT.

The setter here is internal and so can only be called by this contract.  An event is emitted with the updated price multiple information. Also creates a getter to view the current price multiple.

In a followup PR, a public setter will be exposed that enforces atomic setting of the underlying token and the price multiplier.

### CODE REVIEW

File: contracts/Market.sol [hunk 1]
```diff
@@ -155,6 +160,12 @@ contract Market is
    */
   event PriorityRestrictedThresholdSet(uint256 threshold);
 
+  /**
+   * @notice Emitted on setting of `_priceMultiple`.
+   * @param priceMultiple The updated price multiple.
+   */
+  event SetPriceMultiple(uint256 priceMultiple);
```
Comment (scottynomad) [inline contracts/Market.sol:167]: PriceMultipleSet?

File: deploy/deploy-market.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const deploy: DeployFunction = async (environment) => {
     hre,
     feeWallet,
     feePercentage: 15,
+    priceMultiple: 1,
```
Comment (scottynomad) [inline deploy/deploy-market.ts:23]: I think we need to have control down to at least a two decimal places.

**Review by scottynomad — CHANGES_REQUESTED** [review]

File: contracts/Market.sol [hunk 3]
```diff
@@ -343,6 +357,15 @@ contract Market is
     });
   }
 
+  /**
+   * @notice Sets the price multiple, which is the number of base tokens required to purchase one NRT.
```
Comment (jaycenhorton) [inline contracts/Market.sol:361]: > 
```suggestion
   * @notice Set the price multiple, which is the number of base tokens required to purchase one NRT.
```

File: contracts/Market.sol [hunk 4]
```diff
@@ -636,6 +659,14 @@ contract Market is
     }
   }
 
+  /**
+   * @notice Returns the current value of the price multiple, which is the number of base tokens required to
```
Comment (jaycenhorton) [inline contracts/Market.sol:663]:
```suggestion
   * @notice Get the current value of the price multiple, which is the number of base tokens required to
```

File: test/Market.t.sol [hunk 5]
```diff
@@ -861,3 +862,21 @@ contract Market_getRemovalIdsForSupplier is UpgradeableMarket {
...
 }
+
+contract Market__setPriceMultiple is NonUpgradeableMarket {
+  function test() external {
+    vm.recordLogs();
+    uint256 newPriceMultiple = 20;
+    _setPriceMultiple({priceMultiple: newPriceMultiple});
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    assertEq(entries.length, 1);
+    assertEq(entries[0].topics[0], keccak256("SetPriceMultiple(uint256)"));
```
Comment (jaycenhorton) [inline test/Market.t.sol:873]: 💯

File: deploy/deploy-market.ts (hunk 2 above)
Comment (amiecorso) [reply deploy/deploy-market.ts:23]: meaning we want to be able to set a price like $20.34 / NRT? 

to achieve this I guess we just need our price multiple to be scaled by 100 such that 1 is represented by 100 and 20.34 would be represented as 2034 and we divide by 100 after doing the calculations, and just make it very clear how this works in the documentation.  sound right?

File: contracts/Market.sol (hunk 1 above)
Comment (amiecorso) [reply contracts/Market.sol:167]: so Jaycen had asked me to change the tense of another event to match the active voice used in all the OZ dependencies (for example, `TransferBatch`, and now we have `SetHoldbackPercentage`)... what do you guys think?  I realize that it is not consistent with some of our other event names and personally I find it less intuitive in the active voice than in the past tense, but i'm fine with whatever approach

Comment (scottynomad) [reply contracts/Market.sol:167]: Hmm, yeah I would rather have our names be internally consistent since these will be with us forever but not a big deal.

**Review by scottynomad — APPROVED** [review]

File: contracts/Market.sol (hunk 1 above)
Comment (jaycenhorton) [reply contracts/Market.sol:167]: Fwiw active voice seems to be the standard in most popular contracts. Same goes for documentation

**Pull request merged**